src/ConfigFetcher.php

```diff
@@ -116,7 +176,12 @@ public function fetch($etag)
                     $etag = $response->getHeader(self::ETAG_HEADER)[0];
                 }
 
-                return new FetchResponse(FetchResponse::FETCHED, $etag, $body);
+                $url = "";
```

Comment:
Reusing the $url variable made id hard for me to understand its purpose.
I recommend that we use a new variable to store the base_url read from the json.